### PR TITLE
WSTEAM1-1029 - Form uploading screen

### PR DESCRIPTION
**Pull request by amoore108**
Part of https://jira.dev.bbc.co.uk/browse/WSTEAM1-1029

Overall changes
======
- Adds 'Uploading' screen to UGC Form
- Adds initial screen switching switch/case to allow state updates to change the displayed screen
- Small change to background gradient to allow it to fill the entire screen regardless of form height
- Moves Form header and description elements down into the `<Form />` component and out of the root
- Adds Storybook stories for the 2 current screens

Code changes
======
- Adds `<Uploading />` screen component
- Exposes `FormContext.Consumer` to get access to the context at the root of the page to allow screen switching
- Removes the `derivedHtmlType` logic as this is now determined in the BFF
- Adds `initialScreen` prop for the root page to determine which screen to show initially. More so for Storybook benefits

Testing
======
1. Visit https://wsteam1-1029-form-uploading-screen--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/pages-ugc-page--uploading
2. Confirm the layout meets UX and a11y requirements

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/react';
+import Heading from '#app/components/Heading';
+import Paragraph from '#app/components/Paragraph';
+import styles from './styles';
+
+export default function Uploading() {
```
Comment (karinathomasbbc) [inline ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx:7]: Can this be a fat arrow function (we don't tend to use `function` in Simorgh)?

File: ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  return (
+    <>
+      <Heading level={1} id="content" tabIndex={-1} css={styles.heading}>
+        Uploading
+      </Heading>
+      <Paragraph>Please wait until it is finished.</Paragraph>
```
Comment (karinathomasbbc) [inline ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx:13]: Translations - is this a future piece of work?

**Review by karinathomasbbc — APPROVED** [review]

File: ws-nextjs-app/pages/[service]/send/[id]/send.stories.tsx [hunk 3]
```diff
@@ -31,4 +38,5 @@ export default {
   Component,
 };
 
-export const Example = Component;
+export const Form = () => <Component initialScreen="form" />;
+export const Uploading = () => <Component initialScreen="uploading" />;
```
Comment (karinathomasbbc) [inline ws-nextjs-app/pages/[service]/send/[id]/send.stories.tsx:42]: Is it overkill to have stories for the success & error states?

File: ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx (hunk 2 above)
Comment (amoore108) [reply ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx:13]: Yea it will be a separate task for all areas of the Form.

Comment (karinathomasbbc) [reply ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx:13]: Nice, thanks!

File: ws-nextjs-app/pages/[service]/send/[id]/send.stories.tsx (hunk 3 above)
Comment (amoore108) [reply ws-nextjs-app/pages/[service]/send/[id]/send.stories.tsx:42]: We have separate tasks for the success and error states, so was just going to leave any stories for those individual tickets.

File: ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx (hunk 1 above)
Comment (amoore108) [reply ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx:7]: Yea we have a bit of a mixture in the Next.js app. The Next docs tend to use `export default function` for components, but can change this.

Comment (karinathomasbbc) [reply ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx]: Ach I think we can leave it.

Comment (karinathomasbbc) [reply ws-nextjs-app/pages/[service]/send/[id]/Uploading/index.tsx:7]: No worries, let's leave it as is 👍

**Review by HarveyPeachey — APPROVED** [review]

**Pull request merged**